File: ZCopyUnitTester/Mocks/MockBuilder_IConfirmationChecker.vb
Imports Moq
Imports ZCopy

Partial Public Class MockBuilder

    Public Shared Function CreateIConfirmationChecker(expectedReturn As Boolean) As Mock(Of IConfirmationChecker)
        Dim confirmation As New Mock(Of IConfirmationChecker)()
        confirmation.Setup(Function(x) x.GetConfirmation(It.IsAny(Of String))).Returns(expectedReturn)
        Return confirmation
    End Function
End Class

File: ZCopyUnitTester/Mocks/MockBuilder_IConfirmationChecker.cs
using Moq;
using ZCopy;

 namespace ZCopyUnitTester.Mocks
{
    public partial class MockBuilder
    {
        public static Mock<IConfirmationChecker> CreateIConfirmationChecker(bool expectedReturn)
        {
            Mock<IConfirmationChecker> confirmation = new Mock<IConfirmationChecker>();
            confirmation.Setup(x => x.GetConfirmation(It.IsAny<string>())).Returns(expectedReturn);
            return confirmation;
        }
    }
}
